dev_apps/music-nga/test/marionette/lib/music.js

```diff
@@ -0,0 +1,493 @@
+    return header.findElement(Music.Selector.titleText);
+  },
+
+  get sublistShuffleButton() {
+    return this.client.helper.waitForElement(
+      Music.Selector.sublistShuffleButton);
+  },
+
+  // TODO(gareth): Move this shareMenu stuff into the helper.
+  get shareMenu() {
+    // Switch to the system app first.
+    this.client.switchToFrame();
+    return this.client.helper.waitForElement(Music.Selector.shareMenu);
+  },
+
+  get progressBar() {
+    return this.client.findElement(Music.Selector.progressBar);
+  },
+
+  get pickDoneButton() {
+    return this.client.findElement(Music.Selector.pickDoneButton);
+  },
+
+  get header() {
+    return this.client.findElement(Music.Selector.header);
+  },
+
+  get playerIcon() {
+    return this.client.findElement(Music.Selector.playerIcon);
+  },
+
+  get isPlaying() {
+    return this.playButton.getAttribute('class').indexOf('is-pause') === -1;
+  },
+
+  get songProgress() {
+    return parseFloat(this.progressBar.getAttribute('value'));
+  },
+
+  launch: function() {
+    this.client.switchToFrame();
+    this.client.apps.launch(this.origin);
+    this.client.apps.switchToApp(this.origin);
+    this.client.helper.waitForElement('body');
+  },
+
+  close: function() {
+    this.client.apps.close(this.origin);
+  },
+
+  switchToMe: function(options) {
+    options = options || {};
+
+    this.client.switchToFrame();
+
+    // Switch to music even when it is in background.
+    // We cannot use switchToApp here, because it will wait
+    // for the app to come to foreground.
+    if (options.background) {
+      var musicFrame = this.client.findElement('iframe[src*="' +
+                                          this.origin + '"]');
+      this.client.switchToFrame(musicFrame);
+    } else {
+      this.client.apps.switchToApp(this.origin);
+    }
+  },
+
+  waitForListEnumerate: function(frameSelector) {
+    var frame = this.client.findElement(frameSelector);
+    assert.ok(frame, 'Frame must be valid.' + frameSelector);
+
+    this.client.waitFor(function() {
+      var selector = frameSelector;
+      return this.client.executeScript(function(selector) {
+        return document.querySelector(selector);
+      }, [selector]);
+    }.bind(this));
+
+    this.client.switchToFrame(frame);
+
+    this.client.waitFor(function() {
+      return this.client.executeScript(function() {
+        return document.readyState === 'complete';
+      });
+    }.bind(this));
+
+    this.client.waitFor(function() {
+      return this.client.executeScript(function() {
+        var el = document.querySelector('#list');
+        return el; //.state === 'complete';
+      });
+    }.bind(this));
+
+    this.switchToMe();
+  },
+
+  waitFinishedScanning: function() {
+    this.client.switchToFrame(this.endpoint);
+    this.client.waitFor(function() {
+      return this.client.executeScript(function() {
+        return window.wrappedJSObject.Database.initialScanComplete === true;
```

Comment:
I'm actually in the process of exposing 2 `Promise` objects on `Database`; `enumerable` and `ready`

I ran into an issue implementing the "open" activity where I tried using the DB before it was ready which led me to the conclusion of just exposing some promises on `Database`.